core/lib/src/config/secret_key.rs

```diff
@@ -178,6 +193,80 @@ impl SecretKey {
     {
         ser.serialize_bytes(&[0; 32][..])
     }
+
+    /// Encrypts the given data.
+    /// Generates a random nonce for each encryption to ensure uniqueness.
+    /// Returns the Vec<u8> of the concatenated nonce and ciphertext.
+    ///
+    /// # Example
+    /// ```rust
+    /// use rocket::config::SecretKey;
+    ///
+    /// let plaintext = "I like turtles".as_bytes();
+    /// let secret_key = SecretKey::generate().unwrap();
+    ///
+    /// let encrypted = secret_key.encrypt(&plaintext).unwrap();
+    /// let decrypted = secret_key.decrypt(&encrypted).unwrap();
+    ///
+    /// assert_eq!(decrypted, plaintext);
+    /// ```
+    pub fn encrypt<T: AsRef<[u8]>>(&self, value: T) -> Result<Vec<u8>, Error> {
```

Comment:
It would be nice to provide a variant that allows the caller to provide associated data.

Reply:
We can implement it in different ways.

- change the current method by adding an optional argument for AAD:
```rust
pub fn encrypt<T: AsRef<[u8]>, A: AsRef<[u8]>>(
    &self,
    value: T,
    aad: Option<A>
) -> Result<Cipher, Error> {
```

- or add separate method with additional argument (and accordingly a similar method
for `decrypt`):
```rust
pub fn encrypt_with_aad<T: AsRef<[u8]>, A: AsRef<[u8]>>(
    &self,
    value: T,
    aad: A
) -> Result<Cipher, Error> {
```

I prefer the second way. Despite requiring more code, it might be clearer for the user.
Which one do you like more?